include/aspect/particle/particle_handler.h

```diff
@@ -237,6 +422,50 @@ namespace aspect
         update_next_free_particle_index();
 
         /**
+         * Transfer particles that have crossed subdomain boundaries to other
+         * processors. The transfer occurs in two steps. As a first step all
```

Comment:
I'd drop the explanation of *how* the function works from the documentation. What's relevant here is *what* the function does. I say that because I think I'll have comments on how this is implemented once I get to this part of the patch :-)